framework/dotnet/BenchPress.TestFramework/MOCK_DELETE_LATER.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
```

Comment:
Rather than mocking the file out directly like this, I would recommend one of two options:
1. Create an interface structure and using a Mocking library such as Moq to create the mocks 
2. Create a simulator that has an in-memory implementation, with logic that works without hitting external dependencies.

2 will take longer and may not be necessary, so while I generally am against mocking, in this case since we plan on changing this code later I think it may be appropriate. Having interfaces also helps us to define what our API will look like on the engine side

Reply:
delete_later file is just a placeholder until we have a functional core engine. when we have that file, I'll delete this file entirely from the repo.

Reply:
I am not a fan of this approach, lets discuss offline